src/components/cc-pricing-header/cc-pricing-header.smart.js

```diff
@@ -48,6 +49,23 @@ defineSmartComponent({
       },
     );
 
+    /**
+     * This `cc-smart-container` is placed around the whole `cc-pricing-page` component.
```

Comment:
This comment actually comes from `cc-pricing-header` because we already used the same logic to handle `zoneId` changes.

Pricing components are really special because we rely on smart component nesting:
- 1 global `cc-smart-container` that provides context for the `cc-pricing-header`, `cc-pricing-estimation`, all `cc-pricing-product` and all `cc-pricing-product-consumption`,
  - it provides the context that needs to be shared between all these components: `zoneId`, `currency`, `apiHost`, `addonFeatures`.
- Inside it, we also have individual `cc-smart-container` wrapping each `cc-pricing-product` / `cc-pricing-product-consumption`.
  - they inherit the context from the global `cc-smart-container` + they add their own param to it: `productId` (and possibly `addonFeatures` if you want to override the global one but we never really do that).

```html
<cc-smart-container context="{{ apiConfig: { ... }, zoneId: 'mtl', currency: 'USD' }}">
  <cc-pricing-page>
    <cc-pricing-header></cc-pricing-header>

    <cc-smart-container context="{ productId: 'node' }">
      <cc-pricing-product mode="runtime" action="add"></cc-pricing-product>
    <cc-smart-container>
    
    ...
  </cc-pricing-page>
</cc-smart-container>
```

To make all `cc-pricing-product`, `cc-pricing-product-consumption` and `cc-pricing-estimation` component fetch the new price system everytime the currency or the zone changes, the `cc-pricing-header` & `cc-pricing-estimation` do as follows:

- listen for currency / zone changes,
- when a change is detected, change their own context (because their closest smart-container) is the global one, common to all pricing components,
- this makes all `cc-pricing-product` & `cc-pricing-product-consumption` fetch the new price system as well because the global context has changed.

This architecture has been designed when we refactored the components for the new website and if I remember well this was designed to make the whole thing very flexible because the components may be used with or without the `cc-pricing-page`, they may be placed anywhere on the page (the pricing estimation is sticky on the right in desktop but sticky bottom when viewport is reduced), the website must be able to remove / add products on the fly (filter component).

I'll try to come up with a better comment to summarize all this but it's tricky.